update "Connecting to network" page (#24893)
ethereum/go-ethereum: docs/_interface/peer-to-peer.md

- `--goerli`, Goerli proof-of-authority test network
- `--sepolia` Sepolia proof-of-work test network

**Note:** network selection is not persisted in the config file. To connect to a pre-defined network
you must always enable it explicitly, even when using the `--config` flag to load other configuration values.
For example:
Providing these flags at startup instructs Geth to connect to the specific public testnet instead of Ethereum Mainnet. Because these are public testnets that have been running for several years, Geth has to download the historical blockchain data from genesis, just the same as for Ethereum Mainnet.

**Note:** network selection is not persisted in the config file. To connect to a pre-defined network you must always enable it explicitly, even when using the `--config` flag to load other configuration values. For example:

```shell 

```sh
# Generate desired config file. You must specify testnet here.
geth --goerli --syncmode "full" ... dumpconfig > goerli.toml

# Start geth with given config file. Here too the testnet must be specified.
geth --goerli --config goerli.toml

```

## How Peers Are Found
## Finding peers

> Geth continuously attempts to connect to other nodes on the network until it has peers. If

Author: ah ok. Yes, IIUC the subprotocol negotiation is part of the RLPx session and therefore post-discovery. Thanks!

## Connectivity problems

There are occasions when Geth simply fails to connect to peers. The common reasons for this are:

- Local time might be incorrect. An accurate clock is required to participate in the Ethereum network. The local clock can be resynchronized using commands such as `sudo ntpdate -s time.nist.gov` (this will vary depending on operating system).

- Some firewall configurations can prohibit UDP traffic. The static nodes feature or `admin.addPeer()` on the console can be used to configure connections manually.

- The public network Geth is connecting to might be deprecated or have a low number of active nodes that are hard to find. In this case, the best action is to switch to an alternative test network.

- Running Geth in [light mode](../_interface/les.md) often leads to connectivity issues because there are few nodes running light servers. There is no easy fix for this except to switch Geth out of light mode.

## Checking Connectivity

> The `net` module has two attributes that enable checking node connectivity from the [interactive Javascript console](../_interface/JavaScript-Console.md). These are `net.listening` which reports whether the Geth node is listening for inbound requests,a nd `peerCount` which returns the number of active peers the node is connected to.

Reviewer: ```suggestion
The `net` module has two attributes that enable checking node connectivity from the [interactive Javascript console](../_interface/JavaScript-Console.md). These are `net.listening` which reports whether the Geth node is listening for inbound requests, and `peerCount` which returns the number of active peers the node is connected to.
```

## Custom Networks

It is often useful for developers to connect to private test networks rather than public testnets or Etheruem mainnet. These sandbox environments allow block creation without competing against other miners, easy minting of test ether and give freedom to break things without real-world consequences. A private network is started by providing a value to `--networkid` that is not used by any other existing public network ([Chainlist](https://chainlist.org)) and creating a custom `genesis.json` file. Detailed instructions for this are available on the [Private Networks page](/docs/interface/private-network).

```shell

geth --networkid="12345" genesis.json

```

## Static nodes

Geth also supports static nodes. Static nodes are specific peers that are always connected to. Geth reconnects to these peers automatically when it is restarted. Specific nodes are defined to be statc nodes by saving their enode addresses to a json file and providing the file path to Gteh at startup:

```shell

> geth <other flags> datadir/geth/static-nodes.json

Reviewer 2: hm does this work? I thought this file has to be stored in `<datadir>/geth/static-nodes.json` and then geth will automatically find it without the user having to pass it as argument.

Author: yes you are right. Thanks for the catch - updating now.

```javascript
[
  "enode://f4642fa65af50cfdea8fa7414a5def7bb7991478b768e296f5e4a54e8b995de102e0ceae2e826f293c481b5325f89be6d207b003382e18a8ecba66fbaf6416c0@33.4.2.1:30303",
  "enode://pubkey@ip:port"
]
```

Static nodes can also be added at runtime in the Javascript console by passing an enode address to `admin.addPeer()`:

```javascript

admin.addPeer("enode://f4642fa65af50cfdea8fa7414a5def7bb7991478b768e296f5e4a54e8b995de102e0ceae2e826f293c481b5325f89be6d207b003382e18a8ecba66fbaf6416c0@33.4.2.1:30303")

```

## Peer limit

It is sometimes desirable to cap the number of peers Geth will connect to in order to limit on the computational and bandwidth cost associated with running a node. By default, the limit is 50 peers, however, this can be updated by passing a value to `--maxpeers`:

```shell

geth <otherflags> --maxpeers 15

```

## Trusted nodes

Geth supports trusted nodes that are always allowed to reconnect, even if the peer limit is reached. They can be added persistently via a config file `<datadir>/geth/trusted-nodes.json` or temporarily using the Javascript console.

The format for the config file is identical to the one used for static nodes.

```shell

> geth <other flags> datadir/geth/trusted-nodes.json

Reviewer 2: Same with this.
